WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -48,6 +51,9 @@ def create
         @competition.errors[:competition_id_to_clone] = "invalid"
       end
     end
+    if current_user.any_kind_of_delegate?
+      @competition.delegates << current_user
+    end
 
     if @competition.errors.size == 0 && @competition.save
```

Comment:
Welp, turns out if you remove this `@competition.errors.size == 0` from the `if` condition, all our tests still pass =(. I'll add a test for that first, and then see if I can refactor this to get rid of it.